x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -567,6 +567,11 @@ const AssistantComponent: React.FC<Props> = ({
     setCurrentConversation,
   });
 
+  const handleOnChatCleared = useCallback(async () => {
+    await onChatCleared();
+    await refetchResults();
```

Comment:
We need to refetch conversations after chat clear to avoid this bug:

https://github.com/elastic/kibana/assets/6935300/8c7d4e0c-4e36-4e9c-94ec-d12bb9aa6890